src/client/renderer/r_array.c

```diff
@@ -245,3 +245,17 @@ void R_ResetArrayState(void) {
 	r_array_state.arrays = arrays;
 }
 
+/**
+ * @brief
+ */
+void R_DrawArrays(GLenum type, GLint start, GLsizei count) {
+
+	if (!r_state.active_program)
+		return;
+
+	// TODO: should we figure a better way to "dirty" the matrices so that the memcmp doesn't run every time?
+	if (r_state.active_program->UseMatrices)
+		r_state.active_program->UseMatrices(&r_view.projection_matrix, &r_view.modelview_matrix, &r_view.texture_matrix);
+
```

Comment:
That seems too micro-optimize for my liking, especially since UseProgram will happen between every major step anyhow and no two entities will use the same matrix. I'll only look into that TODO if it shows to be a performance concern.